lib/volcanic/authenticator/v1/identity.rb

```diff
@@ -147,13 +155,14 @@ def create(name, principal_id, **opts)
         # if source is nil, default value will be taken.
         # if source is empty, it raise IdentityError
         def payload_handler(**opts)
+          source = opts[:source].to_s
```

Comment:
change in behaviour - if source is nil / not provided in opts, it will now not default; previously if it was not provided, it would be set to 'password'. Is this intentional?

Reply:
so previously when accepting nil value for source, it doesnt do anything and accept it as nil. Now auth service default it as 'password'. 

Just notice a bug here. when doing `opts[:source].to_s` it return an empty string if is nil value. This will make the api raise 422:
```
"message": {
        "source": "\"source\" is not allowed to be empty"
    }
```

I will fix this and add few test

Reply:
I updated this to as pervious way, so now if `opts[:source]` nil, it take default value `password`.
For `opts[:secretless]` if not nil and `opts[:source]` is equal to `password`, it return false, else is true.